src/extension.ts

```diff
@@ -1347,13 +1347,30 @@ function find_unused_workspace(): [number, any] {
     return [id, new_work];
 }
 
-// Supplements the GNOME Shell theme with the extension's theme.
+// Supplements the loaded theme with the extension's theme.
 function load_theme(stylesheet: string): string | any {
     try {
-        let theme = Lib.dbg(Me.path + "/" + stylesheet + ".css");
-        setThemeStylesheet(theme);
-        loadTheme();
-        return theme;
+        const themeContext = St.ThemeContext.get_for_stage(global.stage);
+
+        // Get the existing shell theme if exists
+        const existingTheme = themeContext.get_theme();
+
+        // Load the pop theme style sheet
```

Comment:
I would prefer for this comment to read "Load the Pop Shell style sheet", as it's not really the pop theme we're loading here. This is a nitpick and not something I'd say needs to get done before merging, but it would be nice to have.

Reply:
Fixed